src/offset/local/windows_sys/system_time.rs

```diff
@@ -0,0 +1,95 @@
+use std::{fmt, io::Error, ptr, result::Result};
+
+use super::WinFileTime;
+
+use winapi::um::minwinbase::SYSTEMTIME;
+use winapi::um::timezoneapi::*;
+
+use crate::{windows_sys_call, Datelike, NaiveDateTime, Timelike};
+
+pub(crate) struct WinSystemTime {
+    inner: SYSTEMTIME,
+}
+
+impl fmt::Debug for WinSystemTime {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("WinSystemTime")
+            .field("year", &self.inner.wYear)
+            .field("month", &self.inner.wMonth)
+            .field("dayOfWeek", &self.inner.wDayOfWeek)
+            .field("day", &self.inner.wDay)
+            .field("hour", &self.inner.wHour)
+            .field("minute", &self.inner.wMinute)
+            .field("second", &self.inner.wSecond)
+            .field("ms", &self.inner.wMilliseconds)
+            .finish()
+    }
+}
+
+impl WinSystemTime {
+    pub(crate) fn new() -> Self {
+        let st = SYSTEMTIME {
+            wYear: 0,
+            wMonth: 0,
+            wDayOfWeek: 0,
+            wDay: 0,
+            wHour: 0,
+            wMinute: 0,
+            wSecond: 0,
+            wMilliseconds: 0,
+        };
+
+        Self { inner: st }
+    }
+
+    pub(crate) fn from_naive_datetime(dt: &NaiveDateTime) -> Self {
```

Comment:
This should yield a `Result` with the `Error` happening if the values are out of range (I'm thinking about the year value in particular).

Reply:
Do we want to be catching them here or allowing the window's functions to notify us in the `windows_sys_call` macro?

Reply:
I'm not completely sure what you mean. I think we should yield `Result` here -- this function is only called by `naive_to_local()` which can return a `LocalResult::None` in case this function fails.